src/components/VmsList/VmCardList.js

```diff
@@ -79,31 +96,77 @@ const VmCardList = ({ vms, alwaysShowPoolCard, fetchMoreVmsAndPools }) => {
     }
   }, [vms, scrollerRef, sentinelRef, fetchMoreVmsAndPools])
 
+  const columnList = [
+    { id: ICON },
+    {
+      ...SortFields[NAME],
+      sort: true,
+    },
+    {
+      ...SortFields[STATUS],
+      sort: true,
+    },
+    { id: POOL_INFO },
+    {
+      ...SortFields[OS],
+      sort: true,
+    },
+    { id: ACTIONS },
+  ]
+
   return (
-    <div ref={scrollerRef}>
-      <Gallery hasGutter className={style['gallery-container']}>
-        {vmsAndPools.map(entity => (
-          <GalleryItem key={entity.get('id')}>{
+    <div ref={scrollerRef} className={tableView ? style.tableView : ''}>
+      { !tableView && (
+        <Gallery hasGutter className={style['gallery-container']}>
+          {vmsAndPools.map(entity => (
+            <GalleryItem key={entity.get('id')}>{
+            entity.get('isVm')
+              ? <CardVm vm={entity} />
+              : <CardPool pool={entity} />}
+            </GalleryItem>
+          ))}
+        </Gallery>
+      )}
+      {tableView && (
```

Comment:
So why put the table view inside the `VmCardList` instead of closer to `VmsList`?  It would be more logical to have the table view vs card view render happen in `VmsList`.  That would just mean you need to double up on the infinite scroller or to have a TableListOrCardList wrapper...

Reply:
@sjd78 indeed after moving the logic to `VmsList` the code looks much better! Note that after this refactoring the infinite scroller is now also a separate entity that could be replaced with another loader if needed.